collections/aggregate_groups.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Applies the given aggregator to each group in the given Grouping, returning the results together with the respective group keys
+ *
+ * ```ts
+ * import { aggregateGroups } from "./aggregate_groups.ts"
+ * import { assertEquals } from "../testing/asserts.ts";
+ * 
+ * const foodProperties = {
+ *     'Curry': [ 'spicy', 'vegan' ],
+ *     'Omelette': [ 'creamy', 'vegetarian' ],
+ * }
+ * const descriptions = aggregateGroups(foodProperties,
+ *     (acc, current, key, first) => {
+ *         if (first)
+ *             return `${key} is ${current}`
+ *
+ *         return `${acc} and ${current}`
+ *     },
+ *     '',
+ * )
+ *
+ * assetEquals(descriptions === {
+ *     'Curry': 'Curry is spicy and vegan',
+ *     'Omelette': 'Omelette is creamy and vegetarian',
+ * })
+ * ```
+ */
+export function aggregateGroups<T, A>(
+  record: Record<string, Array<T>>,
+  aggregator: (accumulator: A, current: T, key: string, first: boolean) => A,
```

Comment:
The kotlin version has an optional parameter for current which is not set for the first element. I think trabslating that to

```ts
aggregator: (key: string, first: boolean, accumulator: A, current?: T) => A
```

might be appropiate.

Reply:
what is the benefit of removing `current` for the first `element` of the `array`?

Reply:
Oh sorry - I switched the parameters^^ The accumulator is not set for the first element, because nothing has been accumulated yet :-)

My bad